controller/space_iterations_blackbox_test.go

```diff
@@ -0,0 +1,558 @@
+			assert.Equal(t, *c.Data.Attributes.Name, iterationName)
+			assert.Equal(t, *c.Data.Attributes.Description, iterationDesc)
+
+			// create another Iteration with nil description
+			iterationName2 := "Sprint #23"
+			ci = newCreateSpaceIterationPayload(iterationName2, nil)
+			_, c = test.CreateSpaceIterationsCreated(t, svc.Context, svc, ctrl, fxt.Spaces[0].ID, ci)
+			assert.Equal(t, *c.Data.Attributes.Name, iterationName2)
+			assert.Nil(t, c.Data.Attributes.Description)
+		})
+	})
+
+	s.T().Run("failure", func(t *testing.T) {
+		t.Run("missing space", func(t *testing.T) {
+			// given
+			ci := newCreateSpaceIterationPayload("Sprint #21", nil)
+			svc, ctrl := s.SecuredController()
+			// when/then
+			test.CreateSpaceIterationsNotFound(s.T(), svc.Context, svc, ctrl, uuid.NewV4(), ci)
+		})
+
+		t.Run("duplicate iteration", func(t *testing.T) {
+			fxt := tf.NewTestFixture(t, s.DB,
+				tf.Identities(1),
+				tf.Iterations(2, func(fxt *tf.TestFixture, idx int) error {
+					if idx == 1 {
+						fxt.Iterations[idx].MakeChildOf(*fxt.Iterations[0])
+					}
+					return nil
+				}))
+			iterationName := fxt.Iterations[1].Name
+			iterationDesc := "duplicate iteration description"
+			ci := newCreateSpaceIterationPayload(iterationName, &iterationDesc)
+			svc, ctrl := s.SecuredController(*fxt.Identities[0])
+			// when
+			test.CreateSpaceIterationsConflict(t, svc.Context, svc, ctrl, fxt.Spaces[0].ID, ci)
+		})
+
+		t.Run("not authorized", func(t *testing.T) {
+			// given
+			ci := newCreateSpaceIterationPayload("Sprint #21", nil)
+			svc, ctrl := s.UnSecuredController()
+			// when/then
+			test.CreateSpaceIterationsUnauthorized(s.T(), svc.Context, svc, ctrl, uuid.NewV4(), ci)
+		})
+
+		t.Run("forbidden", func(t *testing.T) {
+			ci := newCreateSpaceIterationPayload("Sprint #21", nil)
+			fxt := tf.NewTestFixture(s.T(), s.DB, tf.CreateWorkItemEnvironment(), tf.Identities(2))
+
+			spaceOwner := fxt.Identities[0]
+			otherIdentity := fxt.Identities[1]
+			rootItr := fxt.Iterations[0]
+			p := fxt.Spaces[0]
+
+			svc, ctrl := s.SecuredControllerWithIdentity(spaceOwner)
+
+			// try creating iteration with space-owner. should pass
+			_, c := test.CreateSpaceIterationsCreated(s.T(), svc.Context, svc, ctrl, p.ID, ci)
+			require.NotNil(s.T(), c.Data.ID)
+			require.NotNil(s.T(), c.Data.Relationships.Space)
+			assert.Equal(s.T(), p.ID.String(), *c.Data.Relationships.Space.Data.ID)
+			assert.Equal(s.T(), iteration.StateNew.String(), *c.Data.Attributes.State)
+			assert.Equal(s.T(), "/"+rootItr.ID.String(), *c.Data.Attributes.ParentPath)
+			require.NotNil(s.T(), c.Data.Relationships.Workitems.Meta)
+			assert.Equal(s.T(), 0, c.Data.Relationships.Workitems.Meta[KeyTotalWorkItems])
+			assert.Equal(s.T(), 0, c.Data.Relationships.Workitems.Meta[KeyClosedWorkItems])
+
+			svc, ctrl = s.SecuredControllerWithIdentity(otherIdentity)
+			test.CreateSpaceIterationsForbidden(s.T(), svc.Context, svc, ctrl, p.ID, ci)
```

Comment:
It would be nice to create a golden file for the error here as well:

```go
res, jerrs := test.CreateSpaceIterationsForbidden(s.T(), svc.Context, svc, ctrl, p.ID, ci)
compareWithGoldenAgnostic(t, filepath.Join(s.testDir, "create", "forbidden.payload.res.golden.json"), jerrs)
compareWithGoldenAgnostic(t, filepath.Join(s.testDir, "create", "forbidden.headers.res.golden.json"), res.Header())
```